### ISSUE TITLE
bozon start break

### ISSUE THREAD

**Issue by bastienff** [railsware/bozon#1]
When i try 'bozon start' the console give me this error:

```
D:\Node\testbozon>bozon start
events.js:141
      throw er; // Unhandled 'error' event
      ^

Error: spawn gulp ENOENT
    at exports._errnoException (util.js:870:11)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:178:32)
    at onErrorNT (internal/child_process.js:344:16)
    at nextTickCallbackWith2Args (node.js:442:9)
    at process._tickCallback (node.js:356:17)
    at Function.Module.runMain (module.js:443:11)
    at startup (node.js:139:18)
    at node.js:968:3
```

I tried but couldn't find the problem.

_System: Win7x64, Node4.4.3, Gulp(Global), Coffee-Script(Global), Bozon(Global)_

**Comment by alchapone** [comment]
Looks like there are issue on Windows machines.. Will look into it deeper

**Comment by MTRNord** [comment]
On Ubuntu in Vbox (Host is win7 and Guest is Ubuntu 14.04 Desktop) is the same Error

**Comment by MTRNord** [comment]
After installing gulp manually it hangs on `[22:05:01] No gulpfile found`

**Comment by MTRNord** [comment]
It fails on Ubuntu with `Error: Cannot find module 'bozon/tasks/all'`

**Comment by alchapone** [comment]
@MTRNord have you tried it with recent version 0.2.0 ?

**Comment by MTRNord** [comment]
i tried install trougj npm. npm say it is 0.2.0 but bozon say it is 0.1.0. I will try the github version

**Comment by MTRNord** [comment]
Using the git master it works

**Comment by Lichtjaeger** [comment]
I have opend a PR: #3

**Comment by alchapone** [comment]
@MTRNord could you verify on `v0.3.1`?

**Comment by IrregularShed** [comment]
Sadly this remains broken for me. bozon 0.3.3 is pulled straight from npm, and `bozon version` is showing the wrong version:
![image](https://cloud.githubusercontent.com/assets/190127/15431011/ccbb1990-1e9f-11e6-856d-534acae74aa6.png)
(Windows 10 x64, Node v4.4.4)

I'll see if I can spend some time figuring it out but I'm pretty busy right now :unamused:

**Comment by MTRNord** [comment]
@IrregularShed and @alchapone I added an pull request #4  for fixing `bozon version`

**Comment by jiuyuehe** [comment]
me too 

![image](https://cloud.githubusercontent.com/assets/4385903/15447498/cefbdd98-1f76-11e6-9824-bf2c4721d358.png)

>  win8.1 (64)   bozon 0.3.4  

this is base function ,  please fixed  !

**Comment by Lichtjaeger** [comment]
You have too update your local bozon instalation too. Not only the global. You also need to migrate your project from <=0.3.2 to >=0.3.3 (rename `app/javascripts/browser` to `app/javascripts/renderer`).

I'm on it this weekend.

**Comment by jiuyuehe** [comment]
yes ,i update 0.3.4 
![image](https://cloud.githubusercontent.com/assets/4385903/15447939/3c0247bc-1f84-11e6-814d-0ba595aa0f8b.png)

but  the same problem !  i can't find . is my local gulp install wrong position?

**Comment by jiuyuehe** [comment]
i debug this code  

```
const spawn = require('child_process').spawn;

const gulp = spawn('gulp', ['start']);

gulp.stdout.on('data',(data)=>{
        console.log('stdout: $(data)');
});

gulp.stderr.on('data',(data)=>{
    console.log('stderr:${data}');
});

gulp.on('close',(code)=>{
    console.log('child process exit, ${code}');
})

```

the error is same  `bozon start`

```
$ node app
events.js:141
      throw er; // Unhandled 'error' event
      ^

Error: spawn gulp ENOENT
    at exports._errnoException (util.js:837:11)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:178:32)
    at onErrorNT (internal/child_process.js:344:16)
    at doNTCallback2 (node.js:429:9)
    at process._tickCallback (node.js:343:17)
    at Function.Module.runMain (module.js:477:11)
    at startup (node.js:117:18)
    at node.js:951:3

```

**Comment by Lichtjaeger** [comment]
I think I found it.
Until I'm done, run `gulp start` instead of `bozon start` please. (In your st001 directory)

**Comment by jiuyuehe** [comment]
no ,  under the st001 directory  run 'npm install '  , the gulp  been  installed;

I think  the default `gulpfile.js`  only 'require('bozon/tasks/all')'  has problems;

or the path  of  `gulp`  is wrong!


@Lichtjaeger

**Comment by alchapone** [comment]
@jiuyuehe could you test `bozon start` with `bozon` version from `master` branch?

**Comment by alchapone** [comment]
Addressed in [v0.3.5](https://github.com/railsware/bozon/releases/tag/v0.3.5)
